userinterface/ViewModels/Profile/ProfileListViewModel.cs

```diff
@@ -14,34 +16,51 @@ public partial class ProfileListViewModel : ViewModelBase
         private const int MaxProfileAttempts = 10;
 
         private readonly ObservableCollection<ProfileListElementViewModel> profileItems;
+        private readonly IViewModelFactory viewModelFactory;
+        private readonly BE.ProfilesModel profilesModel;
 
         // Just for optimization, only for cleaning up
         private readonly Dictionary<BE.ProfileModel, ProfileListElementViewModel> profileViewModelCache;
 
-        private BE.ProfilesModel ProfilesModel { get; }
-
-        public ProfileListViewModel(BE.ProfilesModel profiles)
+        public ProfileListViewModel(userspace_backend.BackEnd backEnd, IViewModelFactory viewModelFactory)
         {
-            ProfilesModel = profiles;
+            System.Diagnostics.Debug.WriteLine("ProfileListViewModel: Constructor started");
+
+            this.viewModelFactory = viewModelFactory ?? throw new ArgumentNullException(nameof(viewModelFactory));
+            this.profilesModel = backEnd?.Profiles ?? throw new ArgumentNullException(nameof(backEnd));
+
             profileItems = new ObservableCollection<ProfileListElementViewModel>();
             profileViewModelCache = new Dictionary<BE.ProfileModel, ProfileListElementViewModel>();
 
+            System.Diagnostics.Debug.WriteLine($"ProfileListViewModel: ProfilesModel null check: {ProfilesModel == null}");
```

Comment:
At some point we will add a logger. That logger will have interface and DI etc. so that it can be hooked up to different endpoints for logging - i.e. it may not go to System.Diagnostics.Debug.WriteLine.
You might be able to save yourself some future rewriting if you make a simple ILogger and Logger to wrap System.Diagnostics.Debug.WriteLine right now. Up to you. Can resolve this once read